Setting hibernate logging to debug causes NullPointerException

To quickly see it, it's enough to set this in our test suite:
```
logger.hibernate.name =	org.hibernate
logger.hibernate.level = debug
```
The Exception:
```
Caused by: java.lang.NullPointerException
	at org.hibernate.reactive.persister.collection.impl.ReactiveBasicCollectionPersister.getSQLInsertRowString(ReactiveBasicCollectionPersister.java:141)
	at org.hibernate.persister.collection.AbstractCollectionPersister.logStaticSQL(AbstractCollectionPersister.java:693)
	at org.hibernate.persister.collection.AbstractCollectionPersister.<init>(AbstractCollectionPersister.java:529)
	at org.hibernate.persister.collection.BasicCollectionPersister.<init>(BasicCollectionPersister.java:59)
	at org.hibernate.reactive.persister.collection.impl.ReactiveBasicCollectionPersister.<init>(ReactiveBasicCollectionPersister.java:48)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.hibernate.persister.internal.PersisterFactoryImpl.createCollectionPersister(PersisterFactoryImpl.java:152)
```

It happens because Hibernate ORM log static queries in the `AbstractCollectionPersister` in the constructor.
The log operation calls `getSQLInsertRowString` that it is overriden by our subclasses because we need to parse the generated queries. Because the log is in the constructor, our parser has not been created yet.
See for example: https://github.com/hibernate/hibernate-reactive/blob/f283b6e8e4e0cb59add668ea70620af860924962/hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveOneToManyPersister.java#L144

--- a/hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java
+++ b/hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java
@@ -39,14 +39,16 @@ import static org.hibernate.reactive.util.impl.CompletionStages.voidFuture;
  */
 public class ReactiveBasicCollectionPersister extends BasicCollectionPersister
 		implements ReactiveAbstractCollectionPersister {
-	private final Parameters parameters;
+
+	private Parameters parameters() {
+		return Parameters.instance( getFactory().getJdbcServices().getDialect() );
+	}
 
 	public ReactiveBasicCollectionPersister(Collection collectionBinding,
 											CollectionDataAccess cacheAccessStrategy,
 											PersisterCreationContext creationContext)
 			throws MappingException, CacheException {
 		super( collectionBinding, cacheAccessStrategy, creationContext );
-		this.parameters = Parameters.instance( getFactory().getJdbcServices().getDialect() );
 	}
 
 	public CompletionStage<Void> reactiveInitialize(Serializable key,
@@ -136,27 +138,47 @@ public class ReactiveBasicCollectionPersister extends BasicCollectionPersister
 	}
 
 	@Override
+	protected String generateInsertRowString() {
+		String sql = super.generateInsertRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateUpdateRowString() {
+		String sql = super.generateUpdateRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateDeleteRowString() {
+		String sql = super.generateDeleteRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateDeleteString() {
+		String sql = super.generateDeleteString();
+		return parameters().process( sql );
+	}
+
+	@Override
 	public String getSQLInsertRowString() {
-		String sql = super.getSQLInsertRowString();
-		return parameters.process( sql );
-	}
-
-	@Override
-	public String getSQLDeleteRowString() {
-		String sql = super.getSQLDeleteRowString();
-		return parameters.process( sql );
-	}
-
-	@Override
-	public String getSQLDeleteString() {
-		String sql = super.getSQLDeleteString();
-		return parameters.process( sql );
+		return super.getSQLInsertRowString();
 	}
 
 	@Override
 	public String getSQLUpdateRowString() {
-		String sql = super.getSQLUpdateRowString();
-		return parameters.process( sql );
+		return super.getSQLUpdateRowString();
+	}
+
+	@Override
+	public String getSQLDeleteRowString() {
+		return super.getSQLDeleteRowString();
+	}
+
+	@Override
+	public String getSQLDeleteString() {
+		return super.getSQLDeleteString();
 	}
 
 	@Override

--- a/hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveOneToManyPersister.java
+++ b/hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveOneToManyPersister.java
@@ -43,14 +43,15 @@ import static org.hibernate.reactive.util.impl.CompletionStages.zeroFuture;
 public class ReactiveOneToManyPersister extends OneToManyPersister
 		implements ReactiveAbstractCollectionPersister {
 
-	private final Parameters parameters;
+	private Parameters parameters() {
+		return Parameters.instance( getFactory().getJdbcServices().getDialect() );
+	}
 
 	public ReactiveOneToManyPersister(Collection collectionBinding,
 									  CollectionDataAccess cacheAccessStrategy,
 									  PersisterCreationContext creationContext)
 			throws MappingException, CacheException {
 		super( collectionBinding, cacheAccessStrategy, creationContext );
-		this.parameters = Parameters.instance( getFactory().getJdbcServices().getDialect() );
 	}
 
 	public CompletionStage<Void> reactiveInitialize(Serializable key,
@@ -139,27 +140,47 @@ public class ReactiveOneToManyPersister extends OneToManyPersister
 	}
 
 	@Override
+	protected String generateInsertRowString() {
+		String sql = super.generateInsertRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateUpdateRowString() {
+		String sql = super.generateUpdateRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateDeleteRowString() {
+		String sql = super.generateDeleteRowString();
+		return parameters().process( sql );
+	}
+
+	@Override
+	protected String generateDeleteString() {
+		String sql = super.generateDeleteString();
+		return parameters().process( sql );
+	}
+
+	@Override
 	public String getSQLInsertRowString() {
-		String sql = super.getSQLInsertRowString();
-		return parameters.process( sql );
-	}
-
-	@Override
-	public String getSQLDeleteRowString() {
-		String sql = super.getSQLDeleteRowString();
-		return parameters.process( sql );
-	}
-
-	@Override
-	public String getSQLDeleteString() {
-		String sql = super.getSQLDeleteString();
-		return parameters.process( sql );
+		return super.getSQLInsertRowString();
 	}
 
 	@Override
 	public String getSQLUpdateRowString() {
-		String sql = super.getSQLUpdateRowString();
-		return parameters.process( sql );
+		return super.getSQLUpdateRowString();
+	}
+
+	@Override
+	public String getSQLDeleteRowString() {
+		return super.getSQLDeleteRowString();
+	}
+
+	@Override
+	public String getSQLDeleteString() {
+		return super.getSQLDeleteString();
 	}
 
 	@Override

